test/cmdlineTests/ast_ir_undeployable_contract/output

```diff
@@ -0,0 +1,8 @@
+IR AST:
+null
+Optimized IR AST:
+null
+IR AST:
+null
+Optimized IR AST:
+null
```

Comment:
Can we make the respective functions in ``CompilerStack`` return ``std::optional<Json>`` instead and just *not* add this crap as output in this case (neither in CLI nor in standard-json)?

Reply:
Sure, I guess this should then get an additional changelog entry (compiler feature?) as we're changing the behavior of an outward facing interface?

Reply:
It should get a changelog entry anyways in general, but one is fine.

Reply:
We do this for all other outputs though:
```bash
echo 'interface I {}' |
    solc test.sol \
        --bin \
        --bin-runtime \
        --opcodes \
        --asm \
        --ir \
        --ir-ast-json \
        --ir-optimized \
        --ir-optimized-ast-json \
        --gas \
        --asm-json \
        --abi \
        --hashes
```
```
======= test.sol:I =======
EVM assembly:
null
Gas estimation:
Opcodes:

Binary:

Binary of the runtime part:

IR:

IR AST:
null
Optimized IR:

Optimized IR AST:
null
Function signatures:
Contract JSON ABI
[]
```

I guess it does make some sense to explicitly show empty output rather than not say anything when the output is empty.

TBH I don't have a strong opinion either way, but it seems inconsistent to change this just for these IR outputs.

Reply:
Looking at this output, it may indeed be a bit surprising to see outputs for everything but the `*Ast` stuff.

Reply:
My suggestion would be to leave this change out of the PR and, if we want to do this, do it separately afterwards. Or not. Like I said, I don't mind either, but leaving it inconsistent like this is the worst combination of both options.

Reply:
I left in the optionals for the json output from compiler stack (as they really are only optionally present) but reproduced the output as it was for previous versions for now.

Reply:
A tiny benefit of leaving them in in Standard JSON is that it makes it easier to see that you're actually using the right output name when constructing he input by hand. Since we don't validate output names, I'm often unsure if I got it wrong or if I'm just getting no output.

But yeah, we generally strip `null`, so it's not wrong to do that here. And I guess compiling an undeployable contract with nothing else is not a common thing to do anyway.